ddtrace/contrib/internal/ray/patch.py

```diff
@@ -0,0 +1,439 @@
+            span.set_tag_str("ray.task.submit_status", "error")
+            raise e
+
+
+def traced_submit_job(wrapped, instance, args, kwargs):
+    """Trace job submission. This function is also responsible
+    of creating the root span.
+    It will also inject _RAY_SUBMISSION_ID and _RAY_JOB_NAME
+    in the env variable as some spans will not have access to them
+    trough ray_ctx
+    """
+
+    # Three ways of specifying the job name, in order of precedence:
+    # 1. Metadata JSON: ray job submit --metadata_json '{"job_name": "train.cool.model"}' train.py
+    # 2. Special submission ID format: ray job submit --submission_id "job:train.cool.model,run:38" train.py
+    # 3. Ray entrypoint: ray job submit train_cool_model.py
+    submission_id = kwargs.get("submission_id") or generate_job_id()
+    kwargs["submission_id"] = submission_id
+    entrypoint = kwargs.get("entrypoint", "")
+    job_name = (
+        os.environ.get("DD_SERVICE")
+        or kwargs.get("metadata", {}).get("job_name", "")
+        or get_dd_job_name_from_submission_id(submission_id)
+        or get_dd_job_name_from_entrypoint(entrypoint)
+    )
+
+    # Root span creation
+    job_span = tracer.start_span("ray.job", service=job_name or DEFAULT_JOB_NAME, span_type=SpanTypes.RAY)
+    _inject_ray_span_tags_and_metrics(job_span)
+    job_span.set_tag_str(RAY_SUBMISSION_ID_TAG, submission_id)
+    tracer.context_provider.activate(job_span)
+    start_long_running_job(job_span)
+
+    try:
+        with tracer.trace(
+            "ray.job.submit", service=job_name or DEFAULT_JOB_NAME, span_type=SpanTypes.RAY
+        ) as submit_span:
+            _inject_ray_span_tags_and_metrics(submit_span)
+            submit_span.set_tag_str(SPAN_KIND, SpanKind.PRODUCER)
+            submit_span.set_tag_str("ray.submission_id", submission_id)
+
+            # Inject the context of the job so that ray.job.run is its child
+            env_vars = kwargs.setdefault("runtime_env", {}).setdefault("env_vars", {})
+            _TraceContext._inject(job_span.context, env_vars)
+            env_vars[RAY_SUBMISSION_ID] = submission_id
+            if job_name:
+                env_vars[RAY_JOB_NAME] = job_name
+
+            try:
+                resp = wrapped(*args, **kwargs)
+                submit_span.set_tag_str("ray.job.submit_status", "success")
+                return resp
+            except BaseException:
+                submit_span.set_tag_str("ray.job.submit_status", "error")
+                raise
+    except BaseException as e:
+        job_span.set_tag_str("ray.job.status", "error")
+        job_span.error = 1
+        job_span.set_exc_info(type(e), e, e.__traceback__)
+        job_span.finish()
+        raise e
+
+
+def traced_actor_method_call(wrapped, instance, args, kwargs):
+    """Trace actor method submission, i.e the Actor.func.remote()
+    call
+    """
+    actor_name = instance._ray_actor_creation_function_descriptor.class_name
+    method_name = args[0]
```

Comment:
Will this value always be in args? Can be passed in kwargs? If so we should use get_argument_value

Reply:
addressed in [936e2f8](https://github.com/DataDog/dd-trace-py/pull/14511/commits/936e2f8879995831232f579c7b92f200c53d0e01)